sources/packages/web/src/services/StudentService.ts

```diff
@@ -59,11 +61,26 @@ export class StudentService {
   }
 
   /**
-   * Client method to call in order to update the student
-   * information using the user information.
+   * Use the information available in the authentication token to update
+   * the user and student data currently on DB.
+   * If the user account does not exists an API custom error will be returned
+   * from the API with the error code MISSING_STUDENT_ACCOUNT.
+   * @returns true if the student account was found and updated, otherwise false
+   * if the student account is missing.
    */
-  async synchronizeFromUserInfo(): Promise<void> {
-    return await ApiClient.Students.synchronizeFromUserInfo();
+  async synchronizeFromUserToken(): Promise<boolean> {
+    try {
+      await ApiClient.Students.synchronizeFromUserToken();
+      return true;
+    } catch (error: unknown) {
+      if (
+        error instanceof ApiProcessError &&
+        error.errorType === MISSING_STUDENT_ACCOUNT
```

Comment:
I am not able to follow this.

Reply:
Yes, if the student account is missing we must redirect the user to the _Student Profile_ page. Before there was 2 API call happening, I removed one.